dandiapi/api/tests/test_asset.py

```diff
@@ -538,6 +538,51 @@ def test_asset_create(api_client, user, draft_version, asset_blob):
     assert draft_version.status == Version.Status.PENDING
 
 
+@pytest.mark.parametrize(
+    'path,valid',
+    [
+        ('foo.txt', True),
+        ('/foo', False),
+        ('', False),
+        ('/', False),
+        ('./foo', False),
+        ('../foo', False),
+        ('foo/.', False),
+        ('foo/..', False),
+        ('foo/./bar', False),
+        ('foo/../bar', False),
+        ('foo//bar', False),
+        ('foo\0bar', False),
+        ('foo/.bar', True),
+    ],
+)
+@pytest.mark.django_db
+def test_asset_create_path_validation(api_client, user, draft_version, asset_blob, path, valid):
+    assign_perm('owner', user, draft_version.dandiset)
+    api_client.force_authenticate(user=user)
+
+    metadata = {
+        'schemaVersion': settings.DANDI_SCHEMA_VERSION,
+        'encodingFormat': 'application/x-nwb',
+        'path': path,
+        'meta': 'data',
+        'foo': ['bar', 'baz'],
+        '1': 2,
+    }
+
+    resp = api_client.post(
+        f'/api/dandisets/{draft_version.dandiset.identifier}'
+        f'/versions/{draft_version.version}/assets/',
+        {'metadata': metadata, 'blob_id': asset_blob.blob_id},
+        format='json',
+    )
+
+    if valid:
+        assert resp.status_code == 200, resp.data
+    else:
+        assert resp.status_code != 200, resp.data
```

Comment:
I think making the status code itself a parameter instead of `valid`, and then just turning this line into
```python
assert resp.status_code == expected_status_code, resp.data
```
would be better, because 
1) it strengthens the test by testing for an explicit status code. The endpoint could return a 500 error, and this test would still pass as it is now.
2) It decreases the amount of code paths through the test. Tests should be simple and ideally not have any complex logic that changes their behavior depending on a parameter. If you find yourself writing conditionals like this in tests, I find that it usually means a) the test needs to be parametrized differently (which is the case here), or b) the test needs to be broken up into multiple smaller-scoped tests.

Reply:
Thank you for pointing this out, this makes a lot of sense.